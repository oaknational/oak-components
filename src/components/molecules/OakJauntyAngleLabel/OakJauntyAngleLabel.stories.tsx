import React from "react";
import { Meta, StoryObj } from "@storybook/react";

import { OakJauntyAngleLabel } from "./OakJauntyAngleLabel";

import { OakFlex } from "@/components/atoms";

const meta: Meta<typeof OakJauntyAngleLabel> = {
  component: OakJauntyAngleLabel,
  tags: ["autodocs"],
  title: "components/molecules/OakJauntyAngleLabel",
  args: { label: "Select one answer" },
  decorators: [
    (Story) => (
      <OakFlex $pa={"inner-padding-xl"} $flexDirection={"row"}>
        <Story />
      </OakFlex>
    ),
  ],
};
export default meta;

type Story = StoryObj<typeof OakJauntyAngleLabel>;

export const Default: Story = {
  render: (args) => <OakJauntyAngleLabel {...args} />,
  args: { $background: "bg-decorative1-main" },
};
