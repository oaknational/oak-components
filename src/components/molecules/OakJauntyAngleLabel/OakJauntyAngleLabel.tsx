import React, { ComponentProps } from "react";

import { OakBox } from "@/components/atoms";

export type OakJauntyAngleLabelProps = {
  label: string;
  as?: string | React.ElementType;
  htmlFor?: string;
} & ComponentProps<typeof OakBox>; // Adding div type for OakBox for aria features
export const OakJauntyAngleLabel = (props: OakJauntyAngleLabelProps) => {
  const { label, ...oakBoxProps } = props;
  return (
    <OakBox
      $borderRadius={"border-radius-m"}
      $ph={"inner-padding-xs"}
      $pv={"inner-padding-ssx"}
      $font={["heading-7", "heading-6"]}
      $transform={"rotate(-1.5deg)"}
      {...oakBoxProps}
    >
      {label}
    </OakBox>
  );
};
