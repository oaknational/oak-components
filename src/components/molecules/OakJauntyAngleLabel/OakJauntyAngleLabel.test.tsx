import React from "react";
import "@testing-library/jest-dom";

import { OakJauntyAngleLabel } from "./OakJauntyAngleLabel";

import renderWithTheme from "@/test-helpers/renderWithTheme";

describe("OakjauntyAngleLabel", () => {
  it("matches snapshot", () => {
    const { container } = renderWithTheme(
      <OakJauntyAngleLabel label="Select one answer" />,
    );
    expect(container).toMatchSnapshot();
  });
});
