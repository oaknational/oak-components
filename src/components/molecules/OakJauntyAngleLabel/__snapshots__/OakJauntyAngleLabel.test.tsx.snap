// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OakjauntyAngleLabel matches snapshot 1`] = `
.c0 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
  border-radius: 0.375rem;
  -webkit-transform: rotate(-1.5deg);
  -ms-transform: rotate(-1.5deg);
  transform: rotate(-1.5deg);
  font-family: --var(google-font),Lexend,sans-serif;
  font-weight: 600;
  font-size: 1rem;
  line-height: 1.25rem;
  -webkit-letter-spacing: 0.0115rem;
  -moz-letter-spacing: 0.0115rem;
  -ms-letter-spacing: 0.0115rem;
  letter-spacing: 0.0115rem;
}

@media (min-width:750px) {
  .c0 {
    font-weight: 600;
  }
}

@media (min-width:750px) {
  .c0 {
    font-size: 1.25rem;
  }
}

@media (min-width:750px) {
  .c0 {
    line-height: 1.5rem;
  }
}

@media (min-width:750px) {
  .c0 {
    -webkit-letter-spacing: 0.0115rem;
    -moz-letter-spacing: 0.0115rem;
    -ms-letter-spacing: 0.0115rem;
    letter-spacing: 0.0115rem;
  }
}

<div>
  <div
    class="c0"
  >
    Select one answer
  </div>
</div>
`;
